### ISSUE TITLE
​Support Mica/Acrylic on Windows

### ISSUE THREAD

**Issue by Daydreamer-riri** [toeverything/AFFiNE#2399]
### Description

Electron has just merged a PR that adds support for Windows Mica material. I think AFFiNE can follow up on this feature, and I am willing to assist in its implementation. 

It is currently unknown when this feature will be released in Electron, but I will continue to keep an eye on it.

### Anything else?

see: 
  - https://github.com/electron/electron/blob/main/docs/api/browser-window.md#winsetbackgroundmaterialmaterial-windows
  - https://learn.microsoft.com/en-us/windows/apps/design/style/mica

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by pengx17** [comment]
AFAIK the change of https://github.com/electron/electron/blob/main/docs/api/browser-window.md#winsetbackgroundmaterialmaterial-windows is pretty new and will land in Electron 26

**Comment by pengx17** [comment]
Yeah, it is now released in https://github.com/electron/electron/releases/tag/v24.4.0
@Daydreamer-riri you can take your time and create a pr for this one.

**Comment by Daydreamer-riri** [comment]
Sure, I will submit this PR as soon as possible.

**Comment by Daydreamer-riri** [comment]
Currently, there are still some issues with this feature, it did not achieve the expected effect when `titleBarStyle` was set to `hidden`. 

I have already provided feedback in the [Issue](https://github.com/electron/electron/issues/38454). I will continue to follow up on it.

**Comment by Daydreamer-riri** [comment]
![image](https://github.com/toeverything/AFFiNE/assets/70067449/8729e1a2-778f-4861-8dc1-275d9de2652e)

Personally, I think the default title bar of Win11 looks pretty good (although there are some details that need to be addressed). However, there is currently a bug where it cannot be resized.

**Comment by pengx17** [comment]
@hwangdev97 can you provide your feedback in designers' perspective?

**Comment by Daydreamer-riri** [comment]
I think perhaps we should wait and see how Electron handles this bug.

**Comment by hwangdev97** [comment]
From a product design perspective, we will provide settings to enable or disable the default title bar, which seems to look good on Windows 11, although some details may need to be improved. I think we can handle this feature after fixing bugs in Electron. Once the new settings are completed, we can provide an option to enable the title bar (while also addressing the details in the screenshot).

**Comment by pengx17** [comment]
@Daydreamer-riri can you please create a PR so that we can have a look?

**Comment by Daydreamer-riri** [comment]
> @Daydreamer-riri can you please create a PR so that we can have a look?

Currently, I just set the backgroundMaterial and transparent values, no further changes have been made.

I think I need to know whether the transparent attribute needs to be `true` after electron fix this bug.

**Comment by pengx17** [comment]
blocked by https://github.com/electron/electron/issues/38454

**Comment by forehalo** [comment]
@pengx17 followup?
